### PR TITLE
Added float16 and bfloat16 support for TripletSemiHardLoss, TripletHardLoss and LiftedStructLoss

### PR DESCRIPTION
**Pull request by HauserA**
**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- TensorFlow version and how it was installed (source or binary): 2.2.0-r3
- TensorFlow-Addons version and how it was installed (source or binary): 0.10.0-dev
- Python version: 3.6
- Is GPU used? (yes/no): yes

**Describe the bug**

When I build a model with mixed precision the input to TripletSemiHardLoss, TripletHardLoss and LiftedStructLoss should accept: float32, float16 or bfloat16.

float32 works fine, but float16 and bfloat16 are not supported by these losses.

**Code to reproduce the issue**
https://colab.research.google.com/drive/1BvOBlBcWnzvKKM5vp8RhYTOK-KMeygWy

```
        loss = tfa.losses.triplet_hard_loss(labels, logits)
    /usr/local/lib/python3.6/dist-packages/tensorflow_addons/losses/triplet.py:174 triplet_hard_loss  *
        pdist_matrix = metric_learning.pairwise_distance(embeddings, squared=True)
    /usr/local/lib/python3.6/dist-packages/tensorflow_addons/losses/metric_learning.py:55 pairwise_distance  *
        pairwise_distances = tf.math.multiply(
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/util/dispatch.py:180 wrapper  **
        return target(*args, **kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/math_ops.py:381 multiply
        return gen_math_ops.mul(x, y, name)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_math_ops.py:6092 mul
        "Mul", x=x, y=y, name=name)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py:506 _apply_op_helper
        inferred_from[input_arg.type_attr]))

    TypeError: Input 'y' of 'Mul' Op has type float32 that does not match type float16 of argument 'x'.
```

**Changed Files**

Added support for float16 and bfloat16 the same way as it is done for tf.nn.softmax_cross_entropy_with_logits:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L3507-L3510

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Thank you for the pull request! Could you add very simple tests? Those are in the tests/ directory.

File: tensorflow_addons/losses/tests/lifted_test.py [hunk 1]
```diff
@@ -100,6 +108,44 @@ def test_lifted_struct():
...
+    labels = np.random.randint(0, num_classes, size=num_data).astype(np.float16)
+
+    loss_np = lifted_struct_loss_np(labels, embedding, margin)
+
+    # Compute the loss in TF.
+    y_true = tf.constant(labels)
+    y_pred = tf.constant(embedding, dtype=tf.bfloat16)
+    cce_obj = lifted.LiftedStructLoss()
+    loss = cce_obj(y_true, y_pred)
+    np.testing.assert_allclose(loss.numpy(), loss_np, rtol=1e-2, atol=1e-2)
```
Comment (WindQAQ) [inline tensorflow_addons/losses/tests/lifted_test.py:146]: How about using `@pytest.mark.parametrize` to and `assert_allclose_according_to_type` to test against different precisions? It makes less duplicated codes :-)

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/tests/kappa_loss_test.py#L70
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/test_utils.py#L221

Comment (HauserA) [reply tensorflow_addons/losses/tests/lifted_test.py:146]: Yes, makes it a lot cleaner.

`assert_allclose_according_to_type` does not check for bfloat16. Which rtol and atol is best used for bfloat16? The triplet loss test with bfloat16 fails for rtol=1e-3 and atol=1e-3 and accepts for rtol=1e-2 and atol=1e-2.

Comment (WindQAQ) [reply tensorflow_addons/losses/tests/lifted_test.py:146]: Hi, I modify `assert_allclose_according_to_type` to make it support bfloat16. See #1698. You could go ahead after it gets merged. Thank you :-)

Comment (HauserA) [reply tensorflow_addons/losses/tests/lifted_test.py]: Ok, thank you :)

**Pull request merged**